src/regdesk/label_builder.js

```diff
@@ -0,0 +1,197 @@
+import { LabelEpl } from 'WebZLP/src/LabelEpl'
+
+/**
+ * Line offset record
+ */
+class LineOffset {
+
+  /**
+   * Initializes a new instance of the LineOffset class.
+   * @param {string} alignmentPoint - Where to align the text from. 'center', 'left', or 'right'.
+   * @param {boolean} isBold - Whether the text should be bold.
+   * @param {number} maxWidth - The maximum width the text can occupy, in pixels. -1 means 'full label width'.
+   * @param {number} maxHeight - The maximum height the text can occupy, in pixels.
+   * @param {number} offsetX - The horizontal canvas offset to the alignment point. -1 means 'centered'.
+   * @param {number} offsetY - The vertical canvas offset to the alignment point.
+   */
+  constructor (
+    alignmentPoint,
+    isBold,
+    maxWidth,
+    maxHeight,
+    offsetX,
+    offsetY
+  ) {
+    if (!["center", "left", "right"].includes(alignmentPoint)) {
+      throw new Error(`Invalid alignment point ${alignmentPoint}.`);
+    }
+    this.alignmentPoint = alignmentPoint;
+
+    this.isBold = isBold;
+    this.maxHeight = maxHeight;
+    this.maxWidth = maxWidth;
+    this.offsetX = offsetX;
+    this.offsetY = offsetY;
+  }
+}
+
+/**
+ * Class for creating a badge label.
+ */
+export class BadgeLabelBuilder {
+
+  /**
+   * Get the font family the label should be using.
+   */
+  static get fontFamily() { return "twemoji, sans-serif"; }
+
+  /**
+   * Get the line offset information for lines on a badge.
+   */
+  static get lineOffsets() {
+    return {
+      // Manually calculated offsets to figure out a good mapping for a label
+      // with a width of 589px and height of 193px.
+      // These will absolutely need to be recalculated for other sizes.
+      // maxX of -1 means the label width
+      // offsetX of -1 means center it
+      "line1": new LineOffset("center", true, -1, 110, -1, 50),
```

Comment:
I have a vague memory in the back of my mind of some type of magical JS concept that was a type of flag, equal only to itself. I can't for the life of me find it right now. It'd be useful here instead of the more verbose enum pattern.

Looks like I can't define const public fields so getters it is.

Reply:
I didn't add the comments as I specifically wanted to keep the line from getting too long, expecting the IDE anyone is using to help guide them. I could switch it to using a parameter object with naming instead.

I've always found inline comments to be difficult to parse so I never think to try them.